src/nvim/tui/tui.h

```diff
@@ -5,6 +5,18 @@
 
 typedef struct TUIData TUIData;
 
+typedef enum {
+  kDecModeSynchronizedOutput = 2026,
+} TerminalDecMode;
+
+typedef enum {
+  kTerminalModeNotRecognized = 0,
+  kTerminalModeSet = 1,
+  kTerminalModeReset = 2,
+  kTerminalModePermanentlySet = 3,
+  kTerminalModePermanentlyReset = 4,
+} TerminalModeState;
```

Comment:
internally we are pretty consistent about using `Term` for TTY-related names, and `terminal*` for embedded `:terminal`. Example: https://github.com/neovim/neovim/blob/7ce36b7997bf04b6fafdfec482b1a29629b49bd8/src/nvim/terminal.h#L21

also, is "Mode State" somewhat redundant?

```suggestion
} TermModeState;
```

Reply:
Would these definitions be useful for `:terminal` in the future, e.g. when `:terminal` handles a child request (the inverse of what TUI is doing here)? Should these live in a header that is shared by both tui.c and terminal.c?

I guess libvterm should abstract that for us, so maybe not.

Reply:
Happy to rename or move definitions around in the future if we ever need/want to, but since I'll leave as-is for now.